Prometheus receiver panics if no configuration is given

### Component(s)

receiver/prometheus

### What happened?

## Description
Run the OpenTelemetry Collector with a prometheus receiver with no configuration settings. The collector panics.

## Steps to Reproduce

## Expected Result
No panic - an error with a meaningful message and a exit code 1.

## Actual Result
Panics, see output


### Collector version

0.66.0

### Environment information

## Environment
OS: Mac OS X


### OpenTelemetry Collector configuration

```yaml
receivers:
    prometheus:
    otlp:
        protocols:
            grpc:
              endpoint: 0.0.0.0:4317

exporters:
    logging:

processors:
    batch:

extensions:
    health_check:
      endpoint: 0.0.0.0:13133
    pprof:
      endpoint: :1888
    zpages:
      endpoint: :55679

service:
    extensions: [pprof, zpages, health_check]
    pipelines:
      metrics:
        receivers: [prometheus]
        processors: [batch]
        exporters: [logging]
```


### Log output

```shell
2022-11-30T02:35:13.837Z        info    prometheusreceiver@v0.66.0/metrics_receiver.go:254      Starting discovery manager      {"kind": "receiver", "name": "prometheus", "pipeline": "metrics"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2dfe84d]

goroutine 1 [running]:
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver.gcInterval(...)
        /builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver@v0.66.0/metrics_receiver.go:302
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver.(*pReceiver).initPrometheusComponents(0xc0010078c0, {0x590d698?, 0xc001372300?}, {0x59111e0?, 0xc000fa26a8}, {0x58cd140, 0xc0007331c0?})
        /builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver@v0.66.0/metrics_receiver.go:273 +0x16d
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver.(*pReceiver).Start(0xc0010078c0, {0x4f1c2f0?, 0xc000134000?}, {0x59111e0, 0xc000fa26a8})
        /builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver@v0.66.0/metrics_receiver.go:89 +0xce
go.opentelemetry.io/collector/service/internal/pipelines.(*Pipelines).StartAll(0xc000f55630, {0x590d6d0, 0xc000134000}, {0x59111a8?, 0xc000fdb7a0})
        /builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/go.opentelemetry.io/collector@v0.66.0/service/internal/pipelines/pipelines.go:103 +0x452
go.opentelemetry.io/collector/service.(*service).Start(0xc0000d1680, {0x590d6d0, 0xc000134000})
```


### Additional context

_No response_

--- a/.chloggen/fix_promreceiver-panic-on-startup.yaml
+++ b/.chloggen/fix_promreceiver-panic-on-startup.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: prometheusreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix bug in prometheus receiver that panics if no configuration is given.
+
+# One or more tracking issues related to the change
+issues: [16538]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/receiver/prometheusreceiver/config_test.go
+++ b/receiver/prometheusreceiver/config_test.go
@@ -70,7 +70,7 @@ func TestLoadTargetAllocatorConfig(t *testing.T) {
 	require.NoError(t, component.UnmarshalConfig(sub, cfg))
 
 	r0 := cfg.(*Config)
-	assert.Nil(t, r0.PrometheusConfig)
+	assert.NotNil(t, r0.PrometheusConfig)
 	assert.Equal(t, "http://localhost:8080", r0.TargetAllocator.Endpoint)
 	assert.Equal(t, 30*time.Second, r0.TargetAllocator.Interval)
 	assert.Equal(t, "collector-1", r0.TargetAllocator.CollectorID)
@@ -81,7 +81,7 @@ func TestLoadTargetAllocatorConfig(t *testing.T) {
 	require.NoError(t, component.UnmarshalConfig(sub, cfg))
 
 	r1 := cfg.(*Config)
-	assert.Nil(t, r0.PrometheusConfig)
+	assert.NotNil(t, r0.PrometheusConfig)
 	assert.Equal(t, "http://localhost:8080", r0.TargetAllocator.Endpoint)
 	assert.Equal(t, 30*time.Second, r0.TargetAllocator.Interval)
 	assert.Equal(t, "collector-1", r0.TargetAllocator.CollectorID)

--- a/receiver/prometheusreceiver/factory.go
+++ b/receiver/prometheusreceiver/factory.go
@@ -18,6 +18,7 @@ import (
 	"context"
 	"errors"
 
+	promconfig "github.com/prometheus/prometheus/config"
 	_ "github.com/prometheus/prometheus/discovery/install" // init() of this package registers service discovery impl.
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer"
@@ -43,7 +44,11 @@ func NewFactory() receiver.Factory {
 }
 
 func createDefaultConfig() component.Config {
-	return &Config{}
+	return &Config{
+		PrometheusConfig: &promconfig.Config{
+			GlobalConfig: promconfig.DefaultGlobalConfig,
+		},
+	}
 }
 
 func createMetricsReceiver(

--- a/receiver/prometheusreceiver/factory_test.go
+++ b/receiver/prometheusreceiver/factory_test.go
@@ -41,6 +41,7 @@ func TestCreateReceiver(t *testing.T) {
 	creationSet := receivertest.NewNopCreateSettings()
 	mReceiver, _ := createMetricsReceiver(context.Background(), creationSet, cfg, nil)
 	assert.NotNil(t, mReceiver)
+	assert.NotNil(t, mReceiver.(*pReceiver).cfg.PrometheusConfig.GlobalConfig)
 }
 
 func TestFactoryCanParseServiceDiscoveryConfigs(t *testing.T) {

